### PR TITLE
sd-dhcp-client: ignore FORCERENEW

### LINKED ISSUE #16774: Talos Security Advisory for Systemd DHCP (TALOS-2020-1142/CVE-2020-13529)
Please review email issued to  systemd-security at redhat dot com regarding this issue

### PR DESCRIPTION
**Pull request by yuwata**
Fixes #16774.

### CODE REVIEW

File: src/libsystemd-network/sd-dhcp-client.c [hunk 1]
```diff
@@ -1863,12 +1863,13 @@ static int client_handle_message(sd_dhcp_client *client, DHCPMessage *message, i
 
                 client_initialize_io_events(client, client_receive_message_udp);
 
-                if (notify_event) {
+                if (IN_SET(client->state, DHCP_STATE_RENEWING, DHCP_STATE_REBINDING) &&
+                    notify_event == SD_DHCP_CLIENT_EVENT_IP_ACQUIRE)
+                        /* TODO: hmm, maybe this is a bug... */
+                        log_dhcp_client(client, "client_handle_ack() returns SD_DHCP_CLIENT_EVENT_IP_ACQUIRE during DHCP client is %s address, ignoring to trigger callback.",
```
Comment (bluca) [inline src/libsystemd-network/sd-dhcp-client.c]: `"client_handle_ack() returned SD_DHCP_CLIENT_EVENT_IP_ACQUIRE while DHCP client was %s the address, skipping callback."`

File: src/libsystemd-network/sd-dhcp-client.c [hunk 2]
```diff
@@ -1587,9 +1587,17 @@ static int client_handle_forcerenew(sd_dhcp_client *client, DHCPMessage *force,
         if (r != DHCP_FORCERENEW)
                 return -ENOMSG;
 
+#if 0
         log_dhcp_client(client, "FORCERENEW");
 
         return 0;
+#else
+        /* Ignore FORCERENEW requests until we implement RFC3118 (Authentication for DHCP Messages)
```
Comment (bluca) [inline src/libsystemd-network/sd-dhcp-client.c]: Maybe add a `FIXME`? Are there issues open to implement these RFCs, to track this?

**Review by bluca — APPROVED** [review]
lgtm, just two minor nits

**Comment by yuwata** [comment]
@bluca Thank you for the suggestions. The above two points are addressed. Upgrading the green label.

**Comment by bluca** [comment]
@yuwata would be good to have github issues to track those new RFCs

**Pull request merged**